<a href="https://colab.research.google.com/github/SDRChumpitaz/predicciones-ventas-comida/blob/main/Limpieza_de_datos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Limpieza de datos

## Lectura de datos

In [203]:
import pandas as pd

In [204]:
file = "/content/drive/MyDrive/Bootcamp - Data Science /Proyecto /sales_predictions.csv"
df = pd.read_csv(file)

In [205]:
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [206]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


Podemos notar que hay datos faltantes y nulos en: "Outlet_Size", "Item_Weight"

In [207]:
df.describe().T

count         mean          std       min  \
Item_Weight                7060.0    12.857645     4.643456     4.555   
Item_Visibility            8523.0     0.066132     0.051598     0.000   
Item_MRP                   8523.0   140.992782    62.275067    31.290   
Outlet_Establishment_Year  8523.0  1997.831867     8.371760  1985.000   
Item_Outlet_Sales          8523.0  2181.288914  1706.499616    33.290   

                                   25%          50%          75%           max  
Item_Weight                   8.773750    12.600000    16.850000     21.350000  
Item_Visibility               0.026989     0.053931     0.094585      0.328391  
Item_MRP                     93.826500   143.012800   185.643700    266.888400  
Outlet_Establishment_Year  1987.000000  1999.000000  2004.000000   2009.000000  
Item_Outlet_Sales           834.247400  1794.331000  3101.296400  13086.964800

## Filtrado de datos

### Filtrado ITEMS

In [208]:
df[["Item_Identifier","Item_Weight"]].sort_values(by ="Item_Identifier")

Item_Identifier  Item_Weight
8043           DRA12        11.60
118            DRA12        11.60
1693           DRA12        11.60
1197           DRA12        11.60
1245           DRA12        11.60
...              ...          ...
7926           NCZ54        14.65
214            NCZ54        14.65
1896           NCZ54          NaN
7166           NCZ54          NaN
7738           NCZ54        14.65

[8523 rows x 2 columns]

Todos los item con el mismo identificador deben pesar lo mismo.

In [209]:
peso_na = df[df["Item_Weight"].isna()]
peso_na

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
7              FDP10          NaN          Low Fat         0.127470   
18             DRI11          NaN          Low Fat         0.034238   
21             FDW12          NaN          Regular         0.035400   
23             FDC37          NaN          Low Fat         0.057557   
29             FDC14          NaN          Regular         0.072222   
...              ...          ...              ...              ...   
8485           DRK37          NaN          Low Fat         0.043792   
8487           DRG13          NaN          Low Fat         0.037006   
8488           NCN14          NaN          Low Fat         0.091473   
8490           FDU44          NaN          Regular         0.102296   
8504           NCN18          NaN          Low Fat         0.124111   

                  Item_Type  Item_MRP Outlet_Identifier  \
7               Snack Foods  107.7622            OUT027   
18              Hard Drinks  113.2834            OUT027   
21             Baking Goods  144.5444            OUT027   
23             Baking Goods  107.6938            OUT019   
29                   Canned   43.6454            OUT019   
...                     ...       ...               ...   
8485            Soft Drinks  189.0530            OUT027   
8487            Soft Drinks  164.7526            OUT027   
8488                 Others  184.6608            OUT027   
8490  Fruits and Vegetables  162.3552            OUT019   
8504              Household  111.7544            OUT027   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
7                          1985      Medium               Tier 3   
18                         1985      Medium               Tier 3   
21                         1985      Medium               Tier 3   
23                         1985       Small               Tier 1   
29                         1985       Small               Tier 1   
...                         ...         ...                  ...   
8485                       1985      Medium               Tier 3   
8487                       1985      Medium               Tier 3   
8488                       1985      Medium               Tier 3   
8490                       1985       Small               Tier 1   
8504                       1985      Medium               Tier 3   

            Outlet_Type  Item_Outlet_Sales  
7     Supermarket Type3          4022.7636  
18    Supermarket Type3          2303.6680  
21    Supermarket Type3          4064.0432  
23        Grocery Store           214.3876  
29        Grocery Store           125.8362  
...                 ...                ...  
8485  Supermarket Type3          6261.8490  
8487  Supermarket Type3          4111.3150  
8488  Supermarket Type3          2756.4120  
8490      Grocery Store           487.3656  
8504  Supermarket Type3          4138.6128  

[1463 rows x 12 columns]

Buscamos un metodo que coloque el peso por identificación.

In [210]:
grupo_items = df.groupby(["Item_Identifier"])[["Item_Weight"]].min().reset_index()
grupo_items

Item_Identifier  Item_Weight
0              DRA12       11.600
1              DRA24       19.350
2              DRA59        8.270
3              DRB01        7.390
4              DRB13        6.115
...              ...          ...
1554           NCZ30        6.590
1555           NCZ41       19.850
1556           NCZ42       10.500
1557           NCZ53        9.600
1558           NCZ54       14.650

[1559 rows x 2 columns]

In [211]:
df = pd.merge(df, grupo_items, on='Item_Identifier', how='left')

In [212]:
df.head()

Item_Identifier  Item_Weight_x Item_Fat_Content  Item_Visibility  \
0           FDA15           9.30          Low Fat         0.016047   
1           DRC01           5.92          Regular         0.019278   
2           FDN15          17.50          Low Fat         0.016760   
3           FDX07          19.20          Regular         0.000000   
4           NCD19           8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  Item_Weight_y  
0  Supermarket Type1          3735.1380           9.30  
1  Supermarket Type2           443.4228           5.92  
2  Supermarket Type1          2097.2700          17.50  
3      Grocery Store           732.3800          19.20  
4  Supermarket Type1           994.7052           8.93

Por defecto, me crea otra columna. Así que, revisaré los datos primero y después desecho la columna original.

In [213]:
df[df["Item_Weight_y"].isna()]

Item_Identifier  Item_Weight_x Item_Fat_Content  Item_Visibility  \
927            FDN52            NaN          Regular         0.130933   
1922           FDK57            NaN          Low Fat         0.079904   
4187           FDE52            NaN          Regular         0.029742   
5022           FDQ60            NaN          Regular         0.191501   

         Item_Type  Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
927   Frozen Foods   86.9198            OUT027                       1985   
1922   Snack Foods  120.0440            OUT027                       1985   
4187         Dairy   88.9514            OUT027                       1985   
5022  Baking Goods  121.2098            OUT019                       1985   

     Outlet_Size Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  \
927       Medium               Tier 3  Supermarket Type3          1569.9564   
1922      Medium               Tier 3  Supermarket Type3          4434.2280   
4187      Medium               Tier 3  Supermarket Type3          3453.5046   
5022       Small               Tier 1      Grocery Store           120.5098   

      Item_Weight_y  
927             NaN  
1922            NaN  
4187            NaN  
5022            NaN

In [214]:
df[df["Item_Weight_x"].isna()]

Item_Identifier  Item_Weight_x Item_Fat_Content  Item_Visibility  \
7              FDP10            NaN          Low Fat         0.127470   
18             DRI11            NaN          Low Fat         0.034238   
21             FDW12            NaN          Regular         0.035400   
23             FDC37            NaN          Low Fat         0.057557   
29             FDC14            NaN          Regular         0.072222   
...              ...            ...              ...              ...   
8485           DRK37            NaN          Low Fat         0.043792   
8487           DRG13            NaN          Low Fat         0.037006   
8488           NCN14            NaN          Low Fat         0.091473   
8490           FDU44            NaN          Regular         0.102296   
8504           NCN18            NaN          Low Fat         0.124111   

                  Item_Type  Item_MRP Outlet_Identifier  \
7               Snack Foods  107.7622            OUT027   
18              Hard Drinks  113.2834            OUT027   
21             Baking Goods  144.5444            OUT027   
23             Baking Goods  107.6938            OUT019   
29                   Canned   43.6454            OUT019   
...                     ...       ...               ...   
8485            Soft Drinks  189.0530            OUT027   
8487            Soft Drinks  164.7526            OUT027   
8488                 Others  184.6608            OUT027   
8490  Fruits and Vegetables  162.3552            OUT019   
8504              Household  111.7544            OUT027   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
7                          1985      Medium               Tier 3   
18                         1985      Medium               Tier 3   
21                         1985      Medium               Tier 3   
23                         1985       Small               Tier 1   
29                         1985       Small               Tier 1   
...                         ...         ...                  ...   
8485                       1985      Medium               Tier 3   
8487                       1985      Medium               Tier 3   
8488                       1985      Medium               Tier 3   
8490                       1985       Small               Tier 1   
8504                       1985      Medium               Tier 3   

            Outlet_Type  Item_Outlet_Sales  Item_Weight_y  
7     Supermarket Type3          4022.7636         19.000  
18    Supermarket Type3          2303.6680          8.260  
21    Supermarket Type3          4064.0432          8.315  
23        Grocery Store           214.3876         15.500  
29        Grocery Store           125.8362         14.500  
...                 ...                ...            ...  
8485  Supermarket Type3          6261.8490          5.000  
8487  Supermarket Type3          4111.3150         17.250  
8488  Supermarket Type3          2756.4120         19.100  
8490      Grocery Store           487.3656         12.150  
8504  Supermarket Type3          4138.6128          8.895  

[1463 rows x 13 columns]

Como son códigos únicos, los desecho.

In [215]:
df["Item_Weight_x"] = df["Item_Weight_y"]

In [216]:
df.drop(927, inplace=True)
df.drop(1922, inplace=True)
df.drop(4187, inplace=True)
df.drop(5022, inplace=True)

Verificamos que se cambió, ahora a dropear la columna adicionales.

In [217]:
df.drop(columns=["Item_Weight_y"], inplace=True)

Renombramos la columna

In [218]:
df.rename(columns={"Item_Weight_x": "Item_Weight"}, inplace=True)

In [219]:
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [220]:
df[df["Item_Weight"].isna()]

Empty DataFrame
Columns: [Item_Identifier, Item_Weight, Item_Fat_Content, Item_Visibility, Item_Type, Item_MRP, Outlet_Identifier, Outlet_Establishment_Year, Outlet_Size, Outlet_Location_Type, Outlet_Type, Item_Outlet_Sales]
Index: []

### Filtrado TIENDAS

In [221]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8519 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8519 non-null   object 
 1   Item_Weight                8519 non-null   float64
 2   Item_Fat_Content           8519 non-null   object 
 3   Item_Visibility            8519 non-null   float64
 4   Item_Type                  8519 non-null   object 
 5   Item_MRP                   8519 non-null   float64
 6   Outlet_Identifier          8519 non-null   object 
 7   Outlet_Establishment_Year  8519 non-null   int64  
 8   Outlet_Size                6109 non-null   object 
 9   Outlet_Location_Type       8519 non-null   object 
 10  Outlet_Type                8519 non-null   object 
 11  Item_Outlet_Sales          8519 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 865.2+ KB


In [222]:
df_outles_1 = df[["Outlet_Size","Outlet_Identifier"]]
grupo_tienda = df_outles_1.groupby(["Outlet_Identifier"])[["Outlet_Size"]].min().reset_index()
grupo_tienda

Outlet_Identifier Outlet_Size
0            OUT010         NaN
1            OUT013        High
2            OUT017         NaN
3            OUT018      Medium
4            OUT019       Small
5            OUT027      Medium
6            OUT035       Small
7            OUT045         NaN
8            OUT046       Small
9            OUT049      Medium

Buscamos una relación

In [229]:
df_outles_2 = df.groupby(["Outlet_Type","Outlet_Location_Type","Outlet_Size"])[["Item_Outlet_Sales"]].min().reset_index()
df_outles_2

Outlet_Type Outlet_Location_Type Outlet_Size  Item_Outlet_Sales
0      Grocery Store               Tier 1       Small            33.9558
1  Supermarket Type1               Tier 1      Medium           111.8544
2  Supermarket Type1               Tier 1       Small           101.8674
3  Supermarket Type1               Tier 2       Small           113.8518
4  Supermarket Type1               Tier 3        High            73.2380
5  Supermarket Type2               Tier 3      Medium            69.2432
6  Supermarket Type3               Tier 3      Medium           241.6854

Notamos que en "Outlet_Type":


1.   Grocery Store de Tier 1 solo puede ser "Small"
2.   Supermarket Type1 de Tier 2 solo puede ser "Small"
3.   Supermarket Type1 de Tier 3 solo puede ser "High"
4.   Supermarket Type2 de Tier 3 solo puede ser "Medium"
5.   Supermarket Type3 de Tier 3 solo puede ser "Medium"


In [231]:
for index, row in df.iterrows():
  if (row["Outlet_Type"] == "Grocery Store")& (row["Outlet_Location_Type"] == "Tier 1") :
    df.at[index, "Outlet_Size"] = "Small"
  elif (row["Outlet_Type"] == "Supermarket Type1") & (row["Outlet_Location_Type"] == "Tier 2") :
    df.at[index, "Outlet_Size"] = "Small"
  elif (row["Outlet_Type"] == "Supermarket Type1") & (row["Outlet_Location_Type"] == "Tier 3") :
    df.at[index, "Outlet_Size"] = "High"
  elif (row["Outlet_Type"] == "Supermarket Type2") & (row["Outlet_Location_Type"] == "Tier 3") :
    df.at[index, "Outlet_Size"] = "Medium"
  elif (row["Outlet_Type"] == "Supermarket Type3") & (row["Outlet_Location_Type"] == "Tier 3") :
    df.at[index, "Outlet_Size"] = "Medium"
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [232]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8519 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8519 non-null   object 
 1   Item_Weight                8519 non-null   float64
 2   Item_Fat_Content           8519 non-null   object 
 3   Item_Visibility            8519 non-null   float64
 4   Item_Type                  8519 non-null   object 
 5   Item_MRP                   8519 non-null   float64
 6   Outlet_Identifier          8519 non-null   object 
 7   Outlet_Establishment_Year  8519 non-null   int64  
 8   Outlet_Size                7964 non-null   object 
 9   Outlet_Location_Type       8519 non-null   object 
 10  Outlet_Type                8519 non-null   object 
 11  Item_Outlet_Sales          8519 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 1.1+ MB


In [234]:
df["Outlet_Size"].isna().sum()

555

Notamos que quedan 555 datos que no hay forma de rellenar. Por ese motivo, hago un drop.

In [237]:
valores_no_accesibles= df["Outlet_Size"].isna()
df = df[~valores_no_accesibles]

In [238]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7964 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            7964 non-null   object 
 1   Item_Weight                7964 non-null   float64
 2   Item_Fat_Content           7964 non-null   object 
 3   Item_Visibility            7964 non-null   float64
 4   Item_Type                  7964 non-null   object 
 5   Item_MRP                   7964 non-null   float64
 6   Outlet_Identifier          7964 non-null   object 
 7   Outlet_Establishment_Year  7964 non-null   int64  
 8   Outlet_Size                7964 non-null   object 
 9   Outlet_Location_Type       7964 non-null   object 
 10  Outlet_Type                7964 non-null   object 
 11  Item_Outlet_Sales          7964 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 808.8+ KB


### Adicional

In [227]:
df.describe().T

count         mean          std       min  \
Item_Weight                8519.0    12.875420     4.646098     4.555   
Item_Visibility            8519.0     0.066112     0.051586     0.000   
Item_MRP                   8519.0   141.010019    62.283594    31.290   
Outlet_Establishment_Year  8519.0  1997.837892     8.369105  1985.000   
Item_Outlet_Sales          8519.0  2181.188779  1706.511093    33.290   

                                   25%          50%          75%           max  
Item_Weight                   8.785000    12.650000    16.850000     21.350000  
Item_Visibility               0.026983     0.053925     0.094558      0.328391  
Item_MRP                     93.844900   143.047000   185.676600    266.888400  
Outlet_Establishment_Year  1987.000000  1999.000000  2004.000000   2009.000000  
Item_Outlet_Sales           834.247400  1794.331000  3100.630600  13086.964800

¿Item_Visibility	= 0?

In [228]:
df[df["Item_Visibility"] == 0]

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
3              FDX07       19.200          Regular              0.0   
4              NCD19        8.930          Low Fat              0.0   
5              FDP36       10.395          Regular              0.0   
10             FDY07       11.800          Low Fat              0.0   
32             FDP33       18.700          Low Fat              0.0   
...              ...          ...              ...              ...   
8480           FDQ58        7.315          Low Fat              0.0   
8484           DRJ49        6.865          Low Fat              0.0   
8486           FDR20       20.000          Regular              0.0   
8494           NCI54       15.200          Low Fat              0.0   
8500           NCQ42       20.350          Low Fat              0.0   

                  Item_Type  Item_MRP Outlet_Identifier  \
3     Fruits and Vegetables  182.0950            OUT010   
4                 Household   53.8614            OUT013   
5              Baking Goods   51.4008            OUT018   
10    Fruits and Vegetables   45.5402            OUT049   
32              Snack Foods  256.6672            OUT018   
...                     ...       ...               ...   
8480            Snack Foods  154.5340            OUT019   
8484            Soft Drinks  129.9652            OUT013   
8486  Fruits and Vegetables   46.4744            OUT010   
8494              Household  110.4912            OUT017   
8500              Household  125.1678            OUT017   

      Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
3                          1998         NaN               Tier 3   
4                          1987        High               Tier 3   
5                          2009      Medium               Tier 3   
10                         1999      Medium               Tier 1   
32                         2009      Medium               Tier 3   
...                         ...         ...                  ...   
8480                       1985       Small               Tier 1   
8484                       1987        High               Tier 3   
8486                       1998         NaN               Tier 3   
8494                       2007         NaN               Tier 2   
8500                       2007         NaN               Tier 2   

            Outlet_Type  Item_Outlet_Sales  
3         Grocery Store           732.3800  
4     Supermarket Type1           994.7052  
5     Supermarket Type2           556.6088  
10    Supermarket Type1          1516.0266  
32    Supermarket Type2          3068.0064  
...                 ...                ...  
8480      Grocery Store           459.4020  
8484  Supermarket Type1          2324.9736  
8486      Grocery Store            45.2744  
8494  Supermarket Type1          1637.8680  
8500  Supermarket Type1          1907.5170  

[526 rows x 12 columns]

Además, podemos darnos cuenta que tiene ventas.No podemos considerar datos con visibilidad 0 pero con ventas

In [239]:
valores_raros= df["Item_Visibility"] == 0
df = df[~valores_raros]

In [241]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7467 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            7467 non-null   object 
 1   Item_Weight                7467 non-null   float64
 2   Item_Fat_Content           7467 non-null   object 
 3   Item_Visibility            7467 non-null   float64
 4   Item_Type                  7467 non-null   object 
 5   Item_MRP                   7467 non-null   float64
 6   Outlet_Identifier          7467 non-null   object 
 7   Outlet_Establishment_Year  7467 non-null   int64  
 8   Outlet_Size                7467 non-null   object 
 9   Outlet_Location_Type       7467 non-null   object 
 10  Outlet_Type                7467 non-null   object 
 11  Item_Outlet_Sales          7467 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 758.4+ KB
